### PR TITLE
Feature/algolia instantsearch sync urls

### LINKED ISSUE #202: URL updates based on search filters and queries
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/196

### Description

n/a

### PR DESCRIPTION
**Pull request by mattwendzina**
Closes #202 

## 📝 Description

Enables the url search params to be built and shared as a user refines their search results. It also enables the usage of forward/backward navigation to step through each step of refinement. 

You can read more here - https://www.algolia.com/doc/guides/building-search-ui/going-further/routing-urls/react-hooks/

## ⛳️ Current behavior (updates)

Instantsearch just refines results but does not update the URL.

## 🚀 New behavior

The URL will get updated each time the user does any kind of search refinement.

## 💣 Is this a breaking change (Yes/No):

No

### CODE REVIEW

**Review by stevenbister — CHANGES_REQUESTED** [review]
Really good mate :) A couple of minor comments

File: packages/osc-ecommerce/app/components/InstantSearch/utils/search-routing.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        windowTitle({ query }) {
+            const queryTitle = query ? `Results for "${query}"` : 'Search';
+            return queryTitle;
+        },
+
+        createURL({ qsModule, routeState, location }) {
+            const urlParts = location.href.match(/^(.*?)\/search/);
+            const baseUrl = `${urlParts ? urlParts[1] : ''}/`;
+            const queryParameters: Partial<QueryParameters> = {};
+            console.log('RS', qsModule);
```
[the remark this answers is not in the document]
Comment (mattwendzina) [reply packages/osc-ecommerce/app/components/InstantSearch/utils/search-routing.ts]: Oops! Sorted 😄

File: packages/osc-ecommerce/app/components/InstantSearch/utils/search-routing.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            if (routeState.query) {
+                queryParameters.query = encodeURIComponent(routeState.query as string);
+            }
+            if (routeState.price && (routeState.price as []).length > 0) {
+                if (Array.isArray(routeState.price)) {
+                    queryParameters.price = (routeState.price as string[])
+                        .filter((q) => q)
+                        .map(encodeURIComponent);
+                } else {
+                    console.log('routeState.price', routeState.price);
```
[the remark this answers is not in the document]
Comment (mattwendzina) [reply packages/osc-ecommerce/app/components/InstantSearch/utils/search-routing.ts]: DOne

File: packages/osc-ecommerce/app/components/InstantSearch/utils/search-routing.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            const allAwards = Array.isArray(award)
+                ? (award as string[])
+                : ([award].filter(Boolean) as string[]);
+            const awardingBodies = Array.isArray(awarding_body)
+                ? (awarding_body as string[])
+                : ([awarding_body].filter(Boolean) as string[]);
+            const studyMethods = Array.isArray(study_method)
+                ? (study_method as string[])
+                : ([study_method].filter(Boolean) as string[]);
+            const prices = Array.isArray(price)
```
[the remark this answers is not in the document]
Comment (mattwendzina) [reply packages/osc-ecommerce/app/components/InstantSearch/utils/search-routing.ts]: Yep, good shout. Done 👍

[the remark this answers is not in the document]
Comment (stevenbister) [reply packages/osc-ecommerce/app/components/InstantSearch/utils/search-routing.ts]: Noice

**Review by stevenbister — APPROVED** [review]
LGTM!

**Pull request closed without merging**